### ISSUE TITLE
Certain local variables which should keep the same name are renamed

### ISSUE THREAD

**Issue by Marcono1234** [FabricMC/tiny-remapper#32]
### Tiny Remapper version
b81c1d9

### Description
It appears that under certain circumstances `javac` adds `LocalVariableTable` entries with the same slot and name as an already existing variable. Currently Tiny Remapper renames them so they all have unique names. However, that causes issues for decompilers:
- CFR emits variables of type `void` (see https://github.com/leibnitz27/cfr/issues/150) unless told to ignore the variable names
- Procyon adds separate local variables for them, failing to assign a value to the actual variable should should receive the value

A more detailed description of the class file format can be found in https://github.com/leibnitz27/cfr/issues/150.
Currently https://github.com/FabricMC/yarn is affected by this as well which can be seen when decompiling the method `net.minecraft.command.arguments.SwizzleArgumentType.parse` (Yarn name) of a mapped version of Minecraft 20w18a.

**Comment by liach** [comment]
So this might require mapping out the control flow...

Maybe it can be simple mapping, like just grouping local vars by slot and type, then check if they are effectively the same by looking if the jump node before the end label leads to the start of another label? That heuristic seems to apply to your example in leibnitz27/cfr#150, but I am not sure if this is the way it actually works out.

**Comment by Marcono1234** [comment]
Would it possibly be enough to not try to create unique variable names if they reuse the same slot? (Though I am not very familiar with Java bytecode).
E.g. when remapping the following (assuming the local variables have the name `☃`):
```
void test() {
    {
        String s = "a";
        s.isEmpty();
    }
    {
        String s = "b";
        s.isEmpty();
    }
}
```
Tiny Remapper currently also gives both `s` variables unique names, even though that is not necessary.

**Comment by sfPlayer1** [comment]
Yes I think TR has to reuse the same name for the same (original name, slot) unless it can prove that the control flows are distinct. The latter is not really in scope for TR, I'll see about implementing the other part.

Thanks for the report and analysis!
